packages/gatsby-image/src/index.js

```diff
@@ -293,6 +390,7 @@ class Image extends React.Component {
 
     if (fluid) {
       const image = fluid
+      const imageVariants = fluidImages || [fluid]
```

Comment:
We do need to add `media` to the initial source when there are image variants, and the order of the sources matters as well.

We could still split them up between the first image and the rest, but we'd need to choose between 1) always rendering a `<source>` for the initial image in case there is a `media` element on it, or 2) conditionally rendering the first `<source>` depending on if there is a `media` element.

1) would increase the rendered HTML size for all consumers regardless of if they were utilizing variants, so I'll work towards accomplishing this with 2).

Reply:
Bit late on responding to this but always including the extra source for the initial image isn't that big of a deal is it? A lot of it should compress well which should be a standard practice.

Consider the weight added when gzipped which is the most common compression format. It shouldn't be all that high. this also only affects noscript/critical support as lazy loading support uses isVisible before rendering markup for the main image. Placeholders would be affected though I guess.